packages/language-server/src/completions/template-paths.ts

```diff
@@ -53,21 +54,31 @@ export async function templatePaths(
             node.parent?.childForFieldName('name')?.text === 'block' &&
             cursorNode?.equals(node.namedChildren[1]))
     ) {
-        const workspaceFolderDirectory = documentUriToFsPath(workspaceFolderUri);
-
         const completions: CompletionItem[] = [];
 
+        const nodeStringStart = cursorNode.startPosition.column + 1;
+        const nodeStringEnd = cursorNode.endPosition.column - 1;
+
+        const searchText = getStringNodeValue(cursorNode)
+            .substring(0, position.character - nodeStringStart);
+
         for (const { namespace, directory } of templateMappings) {
-            const templatesDirectory = path.resolve(workspaceFolderDirectory, directory);
+            const templatesDirectory = path.resolve(workspaceFolderDirectory, directory)
 
             for (const twigPath of await getTwigFiles(directory)) {
-                const relativePathToTwigFromTemplatesDirectory = path.relative(templatesDirectory, twigPath);
-                const includePath = path.join(namespace, relativePathToTwigFromTemplatesDirectory);
+                const relativePathToTwigFromTemplatesDirectory = path.relative(templatesDirectory, twigPath)
+                    // fix paths for win32
+                    .replaceAll(path.sep, path.posix.sep);
+                const includePath = path.posix.join(namespace, relativePathToTwigFromTemplatesDirectory)
 
-                completions.push({
-                    label: includePath,
-                    kind: CompletionItemKind.File,
-                });
+                if (searchText === '' || includePath.startsWith(searchText)) {
+                    completions.push({
+                        label: includePath,
+                        kind: CompletionItemKind.File,
+                        // Send insertion range to extension
+                        data: [ nodeStringStart, nodeStringEnd, ],
```

Comment:
Define this somewhere around the `nodeString*` variables:

```ts
const range: Range = {
    start: { line: position.line, character: nodeStringStart },
    end: { line: position.line, character: nodeStringEnd },
};
```

and pass this range inside of the `textEdit` prop

```suggestion
                        textEdit: {
                            range,
                            newText: includePath,
                        },
```